git/dependencies.py

```diff
@@ -0,0 +1,415 @@
+		if (self.config == None):
+			return
+		cleanPath = self.__cleanPath(path)
+		if (cleanPath == '*'):
+			sections = self.config.sections()
+		else:
+			sections = [cleanPath]
+		for p in sections:
+
+			if (self.config.has_option(p, 'os')):
+				filteredOSTypes = generate_os_types(self.config[p]['os'])
+				if (len(osFilter) > 0 and len(set(filteredOSTypes).intersection(osFilter)) == 0):
+					continue
+
+			dependencyPath = os.path.join(self.repositoryPath, p)
+
+			if (not os.path.exists(dependencyPath) and self.__canCreateSymlink(dependencyPath, p)):
+				self.__createSymlink(dependencyPath, p)
+				continue
+			elif(self.__isSymlink(dependencyPath) and self.__canCreateSymlink(dependencyPath, p)):
+				self.__updateSymlink(dependencyPath, p)
+				continue
+			elif(self.__isSymlink(dependencyPath) and not self.__canCreateSymlink(dependencyPath, p)):
+				self.__removeSymlink(dependencyPath)
+
+			d = GitDependenciesRepository(self.config[p]['url'], dependencyPath)
+			if (not os.path.exists(dependencyPath)):
+				print(dependencyPath + ' was not found, cloning into dependency...')
+				if (self.config.has_option(p, 'freezed')):
+					d.clone(self.config[p]['freezed'])
+					d.checkout(self.config[p]['ref'])
+				else:
+					d.clone(self.config[p]['ref'])
+			else:
+				d.setRemoteURL('origin', self.config[p]['url'])
+				if (self._shouldUsePools()):
+					task = self._runGitTask(['pool', 'relink', p, self.config[p]['url']])
+					if (task.exitCode() != 0):
+						print(task.output, file=sys.stderr)
+						sys.exit(task.exitCode())
+
+				print('Updating ' + dependencyPath)
+				if (self.config.has_option(p, 'freezed')):
+					d.fetch('origin', self.config[p]['freezed'])
+					d.checkout(self.config[p]['ref'])
+				else:
+					d.fetch('origin', self.config[p]['ref'])
+					d.checkout(self.config[p]['ref'])
+					d.integrateChanges(self.config[p]['ref'])
+				d.updateSubmodules(recursive)
+
+		for p in sections:
+			dependencyPath = os.path.join(self.repositoryPath, p)
+			if(recursive and not self.__isSymlink(dependencyPath)):
+				d = GitDependenciesRepository(self.config[p]['url'], dependencyPath)
+				d.__loadDependenciesFile()
+				d.updateDependencies('*', recursive)
+
+		for p in sections:
+			dependencyPath = os.path.join(self.repositoryPath, p)
+			if(self.__isSymlink(dependencyPath)):
+				dependencyPath = self.__resolveSymlinkRealPath(dependencyPath)
+			d = GitDependenciesRepository(self.config[p]['url'], dependencyPath)
+			d.__loadDependenciesFile()
+			if self.config.has_option(p, 'command'):
+				print("Run command on dependency ({})".format(p))
+				with ChangeDir(self.repositoryPath):
+					runCommand = self.config[p]['command']
+					if (runCommand.startswith("!sh ")):
+						task = ShellTask(runCommand[4:])
+						task.run()
+					else:
+						task = Task('git')
+						task.run(runCommand.split(' '))
+					if (task.output != ''):
+						print(task.output)
```

Comment:
Yes, it could be. But If the task fails I don't want to stop the updating process itself. This is the reason why I don't want to return immediately.